risc0/zkp/src/prove/soundness.rs

```diff
@@ -0,0 +1,292 @@
+    }
+
+    /// Conjecture 8.4 [BCIKS21]
+    fn e_proximity_gap_conjectured(&self) -> f32 {
+        let c_1 = 1; // first parameter in Proximity Gaps, Conjecture 8.4
+        let c_2 = 1; // second parameter in Proximity Gaps, Conjecture 8.4
+
+        // 1 / (ηρ)^c_1
+        let first_term = 1.0 / (ETA * RHO).powi(c_1);
+
+        //   (l • n)^c_2 / q
+        // = (n_trace_polys • |D|)^c_2 / ext_field_size
+        let second_term =
+            (self.n_trace_polys * self.lde_domain_size).powi(c_2) / self.ext_field_size;
+
+        first_term * second_term
+    }
+
+    fn e_fri_constant(&self, e_proximity_gap: f32) -> f32 {
+        // (w_rap + d - 1/2) * e_proximity_gap
+        let first_term = (self.n_trace_polys + self.d - 0.5) * e_proximity_gap;
+
+        // (2m + 1) * (|D| + 1) * FRI_FOLD * num_folding_rounds
+        // ----------------------------------------------------
+        //                 sqrt(ρ) * |K|
+        let second_term = {
+            let numerator = (2.0 * M + 1.0)
+                * (self.lde_domain_size + 1.0)
+                * (FRI_FOLD * self.num_folding_rounds) as f32;
+            let denominator = RHO.sqrt() * self.ext_field_size;
+            numerator / denominator
+        };
+
+        first_term + second_term
+    }
+
+    fn e_fri(&self, theta: f32, e_proximity_gap: f32) -> f32 {
+        let e_fri_constant = self.e_fri_constant(e_proximity_gap);
+
+        let e_fri_queries = e_fri_queries(theta);
+
+        e_fri_constant + e_fri_queries
+    }
+
+    fn e_ali(&self, l_plus: f32) -> f32 {
+        l_plus * self.n_trace_polys / self.ext_field_size
+    }
+
+    fn e_deep(&self, l_plus: f32) -> f32 {
+        let h_plus = self.trace_domain_size + self.biggest_combo;
+
+        let numerator = self.d * (h_plus - 1.0) + (self.trace_domain_size - 1.0);
+        let denominator = self.ext_field_size - self.trace_domain_size - self.lde_domain_size;
+        l_plus * numerator / denominator
+    }
+
+    fn e_deep_ali(&self, l_plus: f32) -> f32 {
+        self.e_deep(l_plus) + self.e_ali(l_plus)
+    }
+}
+
+#[cfg(test)]
+mod toy_model {
+    use super::*;
+    fn toy_model(coeffs_size: usize) {
+        // e, field extension degree
+        let ext_size = baby_bear::ExtElem::EXT_SIZE;
+        let field_size = baby_bear::P as f32;
+        let ext_field_size = libm::powf(field_size, ext_size as f32);
+        let trace_domain_size = (coeffs_size / ext_size) as f32;
+        let lde_domain_size = trace_domain_size * INV_RATE as f32;
+
+        let num_folding_rounds = num_folding_rounds(coeffs_size, ext_size);
+
+        let params = Params {
+            n_sigma_mem: 0,
+            n_sigma_bytes: 0,
+            n_trace_polys: 1.0,
+            d: 1.0,
+            biggest_combo: 1.0,
+            ext_size: 4,
+            ext_field_size,
+            trace_domain_size,
+            lde_domain_size,
+            num_folding_rounds,
+        };
+
+        pub fn toy_proven(params: Params) -> f32 {
```

Comment:
I'm confused about this part. 
In my mind these are the relevant regimes: 

- proven UDR
- proven LDR
- toy problem conjecture
- conjecture 8.4 & conjecture 2.3

This `toy_proven` thing seems to be a 5th scenario that I think we can drop? 

If I understand correctly, ethSTARK 5.10.1 describes parameteres for the toy problem conjecture, and ethSTARK 5.10.2 describes parameters for proven soundness (no toy problem).